gcc8 compiler error: class-memaccess

I am trying to compile the posix example platform and getting the following error:
```bash
$ make -f examples/Makefile-posix 
  BUILD    x86_64-unknown-linux-gnu
make -j10 -C build/x86_64-unknown-linux-gnu --no-print-directory all
--- SNIP ---

/openthread/examples/../src/core/net/dhcp6_client.cpp: In member function ‘otError ot::Dhcp6::Dhcp6Client::Solicit(uint16_t)’:
/openthread/examples/../src/core/net/dhcp6_client.cpp:390:48: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘class ot::Ip6::MessageInfo’; use assignment or value-initialization instead [-Werror=class-memaccess]
     memset(&messageInfo, 0, sizeof(messageInfo));
                                                ^

/openthread/examples/../src/core/meshcop/dataset.cpp: In member function ‘otError ot::MeshCoP::Dataset::AppendMleDatasetTlv(ot::Message&) const’:
/openthread/examples/../src/core/meshcop/dataset.cpp:492:56: error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘class ot::MeshCoP::DelayTimerTlv’ from an array of ‘const class ot::MeshCoP::Tlv’ [-Werror=class-memaccess]
             memcpy(&delayTimer, cur, sizeof(delayTimer));
                                                        ^
In file included from /openthread/examples/../src/core/meshcop/dataset.hpp:42,
                 from /openthread/examples/../src/core/meshcop/dataset.cpp:37:
/openthread/examples/../src/core/meshcop/meshcop_tlvs.hpp:1234:7: note: ‘class ot::MeshCoP::DelayTimerTlv’ declared here
 class DelayTimerTlv : public Tlv
       ^~~~~~~~~~~~~
/openthread/examples/../src/core/net/dhcp6_server.cpp: In member function ‘otError ot::Dhcp6::Dhcp6Server::SendReply(otIp6Address&, uint8_t*, ot::Dhcp6::ClientIdentifier&, ot::Dhcp6::IaNa&)’:
/openthread/examples/../src/core/net/dhcp6_server.cpp:431:48: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘class ot::Ip6::MessageInfo’; use assignment or value-initialization instead [-Werror=class-memaccess]
     memset(&messageInfo, 0, sizeof(messageInfo));

```

This on Archlinux 4.17.6-1
gcc version: gcc (GCC) 8.1.1 20180531
commit: 98ed5ea2adf9a7e2641e6ac70fb0bc3c2c7fc74e

It seems this is a newish compiler error introduced in gcc8

--- a/.travis.yml
+++ b/.travis.yml
@@ -142,6 +142,16 @@ matrix:
           packages:
             - gcc-7
             - g++-7
+    - env: BUILD_TARGET="posix" CC="gcc-8" CXX="g++-8"
+      os: linux
+      compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - gcc-8
+            - g++-8
     - env: BUILD_TARGET="posix-ncp-spi" VERBOSE=1
       os: linux
       compiler: gcc

--- a/src/core/meshcop/dataset.cpp
+++ b/src/core/meshcop/dataset.cpp
@@ -487,9 +487,7 @@ otError Dataset::AppendMleDatasetTlv(Message &aMessage) const
         else if (cur->GetType() == Tlv::kDelayTimer)
         {
             uint32_t      elapsed = TimerMilli::GetNow() - mUpdateTime;
-            DelayTimerTlv delayTimer;
-
-            memcpy(&delayTimer, cur, sizeof(delayTimer));
+            DelayTimerTlv delayTimer(static_cast<const DelayTimerTlv &>(*cur));
 
             if (delayTimer.GetDelayTimer() > elapsed)
             {

--- a/src/core/thread/network_diagnostic.cpp
+++ b/src/core/thread/network_diagnostic.cpp
@@ -351,8 +351,7 @@ otError NetworkDiagnostic::FillRequestedTlvs(Message &             aRequest,
 
         case NetworkDiagnosticTlv::kLeaderData:
         {
-            LeaderDataTlv tlv;
-            memcpy(&tlv, &netif.GetMle().GetLeaderDataTlv(), sizeof(tlv));
+            LeaderDataTlv tlv(reinterpret_cast<const LeaderDataTlv &>(netif.GetMle().GetLeaderDataTlv()));
             tlv.Init();
             SuccessOrExit(error = aResponse.Append(&tlv, tlv.GetSize()));
             break;

